bird_cloud_gnn/radar_dataset.py

```diff
@@ -197,12 +221,19 @@ def load(self):
         self.max_distance = info["max_distance"]
         self.min_neighbours = info["min_neighbours"]
 
+    def cache_dir(self):
+        if self.data_folder is None:
+            directory = self.save_dir
+        else:
+            directory = os.path.dirname(self.data_folder)
+        return directory
+
     def has_cache(self):
         graph_path = os.path.join(
-            self.data_folder, f"dataset_storage_{self.name}_{self.hash}.bin"
+            self.cache_dir(), f"dataset_storage_{self.name}_{self.hash}.bin"
```

Comment:
Here I think it would be useful to check for `self.cache_dir() is None` and `return False` before other tests.

Reply:
Would it not be easier to do that check in `cache_dir` and just raise an error in that case? As far as I can see it should never be `None` unless `self.save_dir` can be `None` I don't now about `dgl` enough to see if that can be the case

Reply:
I think you're right. This is resolved.